### PR TITLE
interfaces/unity7: support unity messaging menu

### PR DESCRIPTION
**Pull request by jdstrand**
This adds support for the unity messaging menu. Tested with hello-unity snap from edge.

### CODE REVIEW

File: interfaces/builtin/unity7.go [hunk 1]
```diff
@@ -460,22 +502,65 @@ dbus (send)
...
-		connectedPlugAppArmor: unity7ConnectedPlugAppArmor,
-		connectedPlugSecComp:  unity7ConnectedPlugSecComp,
-		reservedForOS:         true,
+type Unity7Interface struct{}
+
+func (iface *Unity7Interface) Name() string {
+	return "unity7"
+}
+
+func (iface *Unity7Interface) String() string {
```
Comment (zyga) [inline interfaces/builtin/unity7.go]: I don't think we need the string function. Interfaces don't surface anywhere that I'm aware of.

File: interfaces/builtin/unity7.go [hunk 2]
```diff
@@ -460,22 +502,65 @@ dbus (send)
...
+	return iface.Name()
+}
+
+func (iface *Unity7Interface) AppArmorConnectedPlug(spec *apparmor.Specification, plug *interfaces.Plug, slot *interfaces.Slot) error {
+	var pathBuf bytes.Buffer
+	// Unity7 will take the desktop filename and convert all '-' (and '.',
+	// but we don't care about that here because the rule above already
+	// does that) to '_'. Since we know that the desktop filename starts
+	// with the snap name, perform this conversion on the snap name.
+	pathBuf.WriteString(strings.Replace(plug.Snap.Name(), "-", "_", -1))
```
Comment (zyga) [inline interfaces/builtin/unity7.go]: I think you can just do `strings.Replace()` here. What is the need for the buffer?

Comment (jdstrand) [inline interfaces/builtin/unity7.go]: Nothing. Got rid of the pathBuf.

File: interfaces/builtin/unity7.go (hunk 1 above)
Comment (jdstrand) [inline interfaces/builtin/unity7.go]: It was in other interfaces that I based this on. I removed it and just call iface.Name() directly within the interface for the error messages.

**Comment by jdstrand** [comment]
@zyga, I addressed your feedback. Thanks for the review! :)

**Pull request merged**